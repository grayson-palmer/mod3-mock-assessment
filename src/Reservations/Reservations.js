import React, { Component } from 'react';
import './Reservations.scss';

export const Reservations = (props) => {
  return props.reservations.map( ({ id, name, date, time, number }) => {
    return (
      <div className='reservation' key={ id }>
        <h3 className='reservation__name'>{ name }</h3>
        <p className='reservation__date'>{ date }</p>
        <p className='reservation__time'>{ time } PM</p>
        <p className='reservation__number'>Number of Guests: { number }</p>
        <button type='button' className='reservation__cancel'>Cancel</button>
      </div>
    )
  })
}