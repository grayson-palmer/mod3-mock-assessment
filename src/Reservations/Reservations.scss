@import '../styles/variables.scss';

.reservation {
  background-color: white;
  border: 0.25em teal solid;
  border-radius: 0.5em;
  color: teal;
  margin: 1em;
  width: 15em;
  &__cancel {
    background-color: teal;
    color: white;
    font-size: 1em;
    margin: 0 0 1em;
    padding: 0.25em;
    width: 80%;
  }
}