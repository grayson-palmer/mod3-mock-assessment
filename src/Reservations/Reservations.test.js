import React from 'react';
import Reservations from './Reservations.js';
import { shallow } from 'enzyme';

describe('Form', () => {
  let wrapper;
  const mockReservationData = [
    {
      id: 1,
      name: 'Christie',
      date: '12/29',
      time: '7:00',
      number: 12,
    }
  ]

  // beforeEach(() => {
  //   wrapper = shallow(<Reservations reservations={mockReservationData} />);
  // })

  it ('should be a snapshot of the form element', () => {
    wrapper = shallow(<Reservations reservations={mockReservationData} />);
    expect(wrapper).toMatchSnapshot();
  })

  
})