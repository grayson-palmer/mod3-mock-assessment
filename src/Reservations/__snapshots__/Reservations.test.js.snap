// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should be a snapshot of the reservation element 1`] = `
<div
  className="reservation"
  key="1"
>
  <h3
    className="reservation__name"
  >
    Christie
  </h3>
  <p
    className="reservation__date"
  >
    12/29
  </p>
  <p
    className="reservation__time"
  >
    7:00
     PM
  </p>
  <p
    className="reservation__number"
  >
    Number of Guests: 
    12
  </p>
  <button
    className="reservation__cancel"
    type="button"
  >
    Cancel
  </button>
</div>
`;
